### PR TITLE
[Doc] add description for config of dynamic partition

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by hellolilyliuyi — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/table_design/dynamic_partitioning.md [hunk 1]
```diff
@@ -35,16 +35,19 @@ PROPERTIES(
...
+| dynamic_partition.time_unit             | Yes      | The time granularity for dynamically created  partitions. It is a required parameter. Valid values are `HOUR`, `DAY`, `WEEK`, `MONTH`, and `YEAR`.The time granularity determines the suffix format for dynamically created partitions.<br/>  - If the value is `DAY`,  the suffix format for dynamically created partitions is yyyyMMdd. An example partition name suffix is `20200321`.<br/>  - If the value is `WEEK`, the suffix format for dynamically created partitions is yyyy_ww, for example `2020_13` for the 13th week of 2020.<br/>  - If the value is `MONTH`, the suffix format for dynamically created partitions is yyyyMM, for example `202003`.<br/> - If the value is `YEAR`, the suffix format for dynamically created partitions is yyyy, for example `2020`. |
```
Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:41]: image
redundant spaces

Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:41]: image
missing space

Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:41]: image
should be <ul> and <li> tags

File: docs/table_design/dynamic_partitioning.md [hunk 2]
```diff
@@ -35,16 +35,19 @@ PROPERTIES(
...
+| dynamic_partition.history_partition_num | No       | The number of historical partitions created by the dynamic partitions mechanism, with a default value of `0`. When the value is greater than 0, historical partitions are created in advance. From V2.5.2, StarRocks supports this parameter.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:47]: image
-> partitioning

Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:47]: image
-> v, in lowecase

File: docs/table_design/dynamic_partitioning.md [hunk 3]
```diff
@@ -35,16 +35,19 @@ PROPERTIES(
...
+| dynamic_partition.start_day_of_month    | No       | When time_unit is MONTH, this parameter is used to specify the first day of each month. Valid values: 1 to 28. 1 means the 1st of every month and 28 means the 28th of every month. The default value is 1, which means that every month starts on the 1st. The first day can not be the 29th, 30th, or 31st.                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (amber-create) [inline docs/table_design/dynamic_partitioning.md:49]: image
incorrect parameter names. Change them to "dynamic_partition.time_unit".

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by amber-create — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1 branch-3.0 branch-2.5

**Pull request merged**